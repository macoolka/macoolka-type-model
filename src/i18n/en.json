{
    "macoolka.data-model.idRepeat": "Id[{name}] repeat in Model[{model}].",
    "macoolka.data-model.idNotFound": "Id not found {name} in {model}",
    "macoolka.data-model.fieldNameRepeat": "Field name [{name}] repeat in Model[{model}].",
    "macoolka.data-model.typeNameRepeat": "Type name [{name}] repeat in Model[{model}].",
    "macoolka.data-model.typeNotFound": "Type  [{name}] not be found in Model[{model}].",
    "macoolka.data-model.MNameable": "nameable",
    "macoolka.data-model.MDocumentable": "documentable",
    "macoolka.data-model.MExportable": "exportable",
    "macoolka.data-model.MValueable": "valueable",
    "macoolka.data-model.MEnumScalar": "enum",
    "macoolka.data-model.MJsonScalar": "json",
    "macoolka.data-model.MStringScalar": "string",
    "macoolka.data-model.MNumberScalar": "number",
    "macoolka.data-model.MIntScalar": "integer",
    "macoolka.data-model.MBooleanScalar": "boolean",
    "macoolka.data-model.MDateTimeScalar": "date",
    "macoolka.data-model.MKindScalar": "kind",
    "macoolka.data-model.MTypeScalar": "type",
    "macoolka.data-model.MTypeUnionScalar": "union",
    "macoolka.data-model.MTypeIntersectionScalar": "intersection",
    "macoolka.data-model.MScalable": "scalable",
    "macoolka.data-model.MVariable": "variable",
    "macoolka.data-model.field": "field",
    "macoolka.data-model.param": "param",
    "macoolka.data-model.MFunctional": "functional",
    "macoolka.data-model.method": "method",
    "macoolka.data-model.staticmethod": "static method",
    "macoolka.data-model.function": "function",
    "macoolka.data-model.constant": "constant",
    "macoolka.data-model.export": "export",
    "macoolka.data-model.MTypeable": "typeable",
    "macoolka.data-model.interface": "interface",
    "macoolka.data-model.class": "class",
    "macoolka.data-model.typealias": "type alias",
    "macoolka.data-model.module": "module"

}