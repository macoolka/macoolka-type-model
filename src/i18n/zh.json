{
    "macoolka.data-model.idRepeat": "Id[{name}]在模块[{model}]中重复.",
    "macoolka.data-model.idNotFound": "模块{model}必须有一个Id字段",
    "macoolka.data-model.fieldNameRepeat": "在模型({model})中字段名称({name})重复.",
    "macoolka.data-model.typeNameRepeat": "在模块({model})中类型名称({name})重复.",
    "macoolka.data-model.typeNotFound": "在模块({model})中类型名称({name})没有定义.",
    "macoolka.data-model.MNameable": "可命名的",
    "macoolka.data-model.MDocumentable": "可文档的",
    "macoolka.data-model.MExportable": "可导出的",
    "macoolka.data-model.MValueable": "可赋值的",
    "macoolka.data-model.MEnumScalar": "枚举",
    "macoolka.data-model.MJsonScalar": "json",
    "macoolka.data-model.MStringScalar": "字符型",
    "macoolka.data-model.MNumberScalar": "数字型",
    "macoolka.data-model.MIntScalar": "整型",
    "macoolka.data-model.MBooleanScalar": "布尔型",
    "macoolka.data-model.MDateTimeScalar": "日期型",
    "macoolka.data-model.MKindScalar": "kind",
    "macoolka.data-model.MTypeScalar": "类型",
    "macoolka.data-model.MTypeUnionScalar": "union",
    "macoolka.data-model.MTypeIntersectionScalar": "intersection",
    "macoolka.data-model.MScalable": "scalable",
    "macoolka.data-model.MVariable": "可做为变脸",
    "macoolka.data-model.field": "字段",
    "macoolka.data-model.param": "参数",
    "macoolka.data-model.MFunctional": "可作为函数",
    "macoolka.data-model.method": "方法",
    "macoolka.data-model.staticmethod": "静态方法",
    "macoolka.data-model.function": "函数",
    "macoolka.data-model.constant": "常量",
    "macoolka.data-model.export": "导出",
    "macoolka.data-model.MTypeable": "可作为类型",
    "macoolka.data-model.interface": "接口",
    "macoolka.data-model.class": "类",
    "macoolka.data-model.typealias": "类型",
    "macoolka.data-model.module": "模块"
}