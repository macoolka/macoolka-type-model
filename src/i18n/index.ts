import en from './en.json'
import zh from './zh.json'
import buildApp, { MessageInfo } from 'macoolka-i18n'
export const defaultOption = {
    defaultLanguage: 'en',
    locale: 'en',
    languages: ['en', 'zh'],
    data: {
        en,
        zh
    }
}
export type Message = MessageInfo<keyof typeof defaultOption.data.en, {
    model: string,
    name: string
}>

export const buildI18N = buildApp<Message>(defaultOption)
export default buildI18N