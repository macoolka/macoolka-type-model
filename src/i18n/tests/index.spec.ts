import i18n from '../'
describe('i18n',()=>{
    it('macoolka.data-model.idRepeat',()=>{
      const result= i18n([{id:'macoolka.data-model.idRepeat',value:{model:'test',name:'name'}}])({})
      expect(result).toMatchSnapshot()
    })
})