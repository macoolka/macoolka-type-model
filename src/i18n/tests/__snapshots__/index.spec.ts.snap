// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n macoolka.data-model.idRepeat 1`] = `"Id[name] repeat in Model[test]."`;
